src/pspm_load_data.m

```diff
@@ -253,15 +253,20 @@
             warning('ID:non_existing_channeltype', 'There are no channels of type ''%s'' in the datafile', chan); return;
         end
 
-        data = data(flag == 1);    
+        data = data(flag == 1); 
+        filestruct.posofchannels = find(flag == 1);
     elseif isnumeric(chan)
         if chan == 0, chan = 1:numel(data); end
         if any(chan > numel(data))
             warning('ID:invalid_input', 'Input channel number(s) are greater than the number of channels in the data'); return;
         end
         data = data(chan);
+        filestruct.posofchannels = chan;
     elseif isstruct(chan) && ~isempty(fn) && (~exist(fn, 'file') || chan.options.overwrite == 1)
         save(fn, 'infos', 'data');
+        filestruct.posofchannels = 1:numel(data);

```

Comment:
Could I know if this sentence means all channels will be used?

Reply:
In this case, pspm_load_data checks & saves data that have been passed over as a struct. It then returns everything that was passed as an input, and so for consistency I wanted to populate the fields anyway.